src/shims/yield.h

```diff
@@ -156,9 +156,9 @@ void *_dispatch_wait_for_enqueuer(void **ptr);
 #elif defined(_WIN32)
 #define _dispatch_preemption_yield(n) { (void)n; Sleep(0); }
 #define _dispatch_preemption_yield_to(th, n) { (void)n; Sleep(0); }
-#else 
-#define _dispatch_preemption_yield(n) { (void)n; pthread_yield(); }
-#define _dispatch_preemption_yield_to(th, n) { (void)n; pthread_yield(); }
+#else
+#define _dispatch_preemption_yield(n) { (void)n; sleep(0); }
+#define _dispatch_preemption_yield_to(th, n) { (void)n; sleep(0); }
```

Comment:
This impacts the Linux and FreeBSD paths which, unlike android, have a reasonable libc, and a working pthread implementation.  I think that you should add an `__ANDROID__` clause here and use `sleep`, as the other environments have `pthread_yield`.

Reply:
Got it, thanks. I’ve updated the patch.